tree/dataframe/inc/ROOT/RDF/RInterface.hxx

```diff
@@ -1885,14 +1882,13 @@ public:
       const bool convertVector2RVec = true;
       const auto isCustom = std::find(customCols.begin(), customCols.end(), column) != customCols.end();
       if (!isCustom) {
-         return RDFInternal::ColumnName2ColumnTypeName(std::string(column), fLoopManager->GetID(),
-                                                       fLoopManager->GetTree(), fLoopManager->GetDataSource(), isCustom,
-                                                       convertVector2RVec);
+         return RDFInternal::ColumnName2ColumnTypeName(std::string(column), fLoopManager->GetTree(),
+                                                       fLoopManager->GetDataSource(), isCustom, convertVector2RVec);
       } else {
          // must convert the alias "__rdf::column_type" to a readable type
          const auto colID = std::to_string(fCustomColumns.GetColumns().at(std::string(column))->GetID());
-         const auto call = "ROOT::Internal::RDF::TypeID2TypeName(typeid(__rdf" + std::to_string(fLoopManager->GetID()) +
-                           "::" + std::string(column) + colID + "_type))";
+         const auto call =
+            "ROOT::Internal::RDF::TypeID2TypeName(typeid(__rdf::" + std::string(column) + colID + "_type))";
```

Comment:
Why the spelling ```__rdf``` ?  Leading underscore is technically reserved for the compiler ...

Reply:
I think all identifiers including double underscores in any position are in principle reserved. ROOT sometimes uses such identifiers anyway for "special" things. I think the choice was made (a couple of years ago) to make sure we never conflict with anything, but still produce readable code (e.g. `ROOTRDataFrameReservedNameSpace` also will not conflict with anything but hurts readabiliy).

I'm open to changes though

Reply:
We have often used ```R__``` for that purpose.

Reply:
But that's just as reserved as far as I know

Reply:
Why?  At least not according to ```http://www.gnu.org/software/libc/manual/html_node/Reserved-Names.html```

Reply:
See https://stackoverflow.com/questions/228783/what-are-the-rules-about-using-an-underscore-in-a-c-identifier and the section of the standard mentioned in the first answer, 17.4.3.1.2 Global names [lib.global.names]

Reply:
They reserved double underscore anywhere in the name!! 

So minor tweak :) ```R_rdf```